doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,952 @@
+ntp server 99.1.1.1 key 1
+ntp trusted-keys 1
+ntp trusted-keys 2 
+!
+
+sonic(config)#ntp server 10.11.0.1
+sonic(config)#ntp server pool.ntp.org
+sonic(config)#ntp source-interface Management 0
+sonic(config)#ntp source-interface Loopback 100 
+sonic(config)#do show running-configuration 
+!
+ntp server 10.11.0.1
+ntp server pool.ntp.org
+ntp source-interface Management 0
+ntp source-interface Loopback 100 
+!
+
+sonic(config)# no ntp source-interface Loopback 100 
+sonic(config)# ntp vrf mgmt 
+sonic(config)# do show running-configuration 
+!
+ntp server 10.11.0.1
+ntp server pool.ntp.org
+ntp source-interface Management 0
+ntp vrf mgmt
+!
+
+sonic(config)# ntp vrf default 
+sonic(config)# do show running-configuration
+!
+ntp server 10.11.0.1
+ntp server pool.ntp.org
+ntp source-interface Management 0
+ntp vrf default
+!
+
+```
+    
+#### 3.6.2.3 Debug Commands
+```
+From KLISH:
+
+show ntp associations
+
+show ntp server
+
+show ntp global
+
+```
+
+```
+From shell:
+
+servcie ntp status
+
+check /etc/ntp.conf
+
+check /var/log/syslog and look for ntp
+
+check "docker exec -it mgmt tail -f /var/log/rest_server/rest_server.log" for rest logs
+
+ifconfig lo
+
+ifconfig lo-m
+
+show mgmt-vrf
+
+ntpq -pn
+
+```
+    
+#### 3.6.2.4 IS-CLI Compliance
+
+### 3.6.3 REST API Support
+```
+GET - Get existing NTP configuration information from CONFIG DB.
+      Get NTP peer states
+PUT  - Create NTP configuration into CONFIG DB. 
+POST - Add NTP configuration into CONFIG DB.
+PATCH - Update existing NTP configuraiton information in CONFIG DB.
+DELETE - Delete a existing NTP configuration from CONFIG DB.
+```
+    
+# 4 Flow Diagrams
+
+# 5 Error Handling
+
+# 6 Serviceability and Debug
+
+# 7 Warm Boot Support
+NA      
+
+# 8 Scalability
+
+# 9 Unit Test
+
+The unit-test for this feature will include:
+#### Configuration via CLI
+
+| Test Name | Test Description |
+| :-------- | :----- |
+| Configure NTP server | Verify NTP servers are installed correctly in the configDB and reflected in the NTP peer status |
+| Delete NTP server | Verify NTP server is deleted from the configDB and reflected in the NTP peer status  |
+| Configure NTP source interface| Verify NTP source interface is installed correctly in the configDB, NTP packets are transmitted and received over this source |
+|                               | Verify that multiple NTP source interfaces can be configured, for both default and mgmt vrf cases|
+| Delete NTP source interface| Verify that NTP source interface is removed from the configDB, NTP packets are transmitted and received over the default interface|
+| Configure NTP vrf| Verify that NTP vrf is installed correctly in the configDB and ntp service is running in the specified VRF|
+|                  | Verify that only default and mgmt can be configured as NTP vrf|
+| Delete NTP vrf| Verify that NTP vrf is removed from the configDB and ntp service is running in the default instance|

```

Comment:
There seems to be some confusion between Section 1.1.1.5 which talks about VRF configuration when Management VRF is configured (#### 1.1.1.5 Overall Behavior related to NTP source interface and NTP vrf     
##### When mgmt VRF is configured  ), and this Unit Test.  Please clarify that this UT is being run when no Management  VRF is configured ? (Reference JIRA ID https://it-jira.broadcom.net/browse/SONIC-33434) (Dell SNC-????)